th2_data_services/provider/v5/provider_api/http.py

```diff
@@ -134,18 +132,25 @@ def get_url_search_sse_events(
         attached_messages: bool = False,
         filters: str = "",
     ) -> str:
+        """REST-API `search/sse/events` call create a sse channel of event metadata that matches the filter.
+
+        https://github.com/th2-net/th2-rpt-data-provider#sse-requests-api
+        """
         kwargs = {
-            "startTimestamp": start_timestamp,
+            "startTimestamp": int(start_timestamp.replace(tzinfo=timezone.utc).timestamp() * 1000),
             "resumeFromId": resume_from_id,
             "parentEvent": parent_event,
             "searchDirection": search_direction,
             "resultCountLimit": result_count_limit,
-            "endTimestamp": end_timestamp,
             "limitForParent": limit_for_parent,
             "keepOpen": keep_open,
             "metadataOnly": metadata_only,
             "attachedMessages": attached_messages,
         }
+
+        if end_timestamp is not None:
+            kwargs["endTimestamp"] = int(end_timestamp.replace(tzinfo=timezone.utc).timestamp() * 1000)
+
```

Comment:
There should be no transformations here. It should be on the shoulders of the API user